File: Reversal algorithm.cpp
void reversearray(int a[],int start, int stop)
{
    while(start < stop)
    {
        int temp = a[start];
        a[start] = a[stop];
        a[stop] = temp;
        start++;
        stop--;
    }
}

void printarray(int a[],int n)
{
    for(int i = 0;i < n;i++)
        cout<<a[i]<<" ";
}

void reversalalgo(int a[],int n, int d)
{
    reversearray(a,0,d-1);
    reversearray(a,d,n-1);
    reversearray(a,0,n-1);
}

int main()
{
    int a[] = {1,2,3,4,5,6,7};
    int n = sizeof(a)/sizeof(a[0]);
    reversalalgo(a,n,2);
    printarray(a,n);
}

File: Reversal algorithm.py

def reversearray(a, start, stop):
    while(start < stop):
        temp = a[start]
        a[start] = a[stop]
        a[stop] = temp
        start += 1
        stop -= 1

def reversalalgo(a, d):
    reversearray(a, 0, d-1)
    reversearray(a, d, len(a) - 1)
    reversearray(a, 0, len(a) - 1)

a = [1, 2, 3, 4, 5, 6, 7]
reversalalgo(a, 2)
print(a)
